README-internal.md

```diff
@@ -1,5 +1,10 @@
 # Working with the WordPress (Composer Managed) upstream
 
+## "Release" and "non-release" commits
+The composer-managed upstreams (including Drupal (Composer Managed)) have a idiosyncratic concept of "release" or "normal" commits and "non-release" commits. **Release** commits are those that affect files that are ultimately pushed to the `pantheon-upstreams` repository. This includes the everything that makes the base upstream work and _excludes_ any CI-related files, scripts or tests. **Non-release** commits are those that affect files that are not pushed to the `pantheon-upstreams` repository. This includes CI-related files, scripts, tests, and any other files that are not part of the base upstream.
+
+Because we tend to prefer Squash merges on PRs rather than Merge commits, it is vital to separate interests and not combine _release_ and _non-release_ commits in the same PR. Doing so will lead to portions of the PR being dropped when the deploy script is run, leading to a mismatch between the `pantheon-systems` and `pantheon-upstreams` repositories.
```

Comment:
Multiple commits in a single PR that keep the division between release and non-release is allowed and encouraged, *but should not be squash merged*.

This could further be addressed either with a PR template that clarifies a workflow, or a comment out of the existing GH action that comments "Hey I'm going to allow this to happen but don't you dare squash merge"

Reply:
the comment could get quite noisy on each commit tho , it would have to edit its own comment on subsequent pushes.

Reply:
Yeah, I will create an issue to extend the Check Commits action to check commits across the PR, too. In the meantime, I can address in the note.

Reply:
https://github.com/pantheon-systems/wordpress-composer-managed/issues/150